packages/web-components/src/tooltip/tooltip.ts

```diff
@@ -0,0 +1,198 @@
+   */
+  @attr
+  public anchor: string = '';
+
+  /**
+   * Reference to the anchor element
+   * @internal
+   */
+  private get anchorElement(): HTMLElement | null {
+    const rootNode = this.getRootNode();
+    return (rootNode instanceof ShadowRoot ? rootNode : document).getElementById(this.anchor ?? '');
+  }
+
+  /**
+   * Reference to the anchor positioning style element
+   * @internal
+   */
+  protected anchorPositioningStyleElement: HTMLStyleElement | null = null;
+
+  public constructor() {
+    super();
+    this.elementInternals.role = 'tooltip';
+  }
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+    if (this.anchorElement) {
+      // @ts-expect-error - Baseline 2024
+      const anchorName = this.anchorElement.style.anchorName || `--${this.anchor}`;
+      this.anchorElement.setAttribute('aria-describedby', this.id);
+
+      if (SUPPORTS_ANCHOR_POSITIONING) {
+        // @ts-expect-error - Baseline 2024
+        this.anchorElement.style.anchorName = anchorName;
+        // @ts-expect-error - Baseline 2024
+        this.style.positionAnchor = anchorName;
+      } else {
+        // Provide style fallback for browsers that do not support anchor positioning
+        if (!this.anchorPositioningStyleElement) {
+          this.anchorPositioningStyleElement = document.createElement('style');
+          document.head.append(this.anchorPositioningStyleElement);
+        }
+
+        // Given a position with <direction>-<alignment> format, return the proper CSS properties
+        const [direction, alignment] = this.positioning?.split('-') ?? [];
+        let centerAlignment;
+
+        if (alignment === undefined) {
+          if (direction === 'above' || direction === 'below') {
+            centerAlignment = 'centerX';
+          }
+          if (direction === 'before' || direction === 'after') {
+            centerAlignment = 'centerY';
+          }
+        }
+
+        const directionCSS = {
+          above: `bottom: anchor(${anchorName} top);`,
+          below: `top: anchor(${anchorName} bottom);`,
+          before: `right: anchor(${anchorName} left);`,
+          after: `left: anchor(${anchorName} right);`,
+        }[direction];
+
+        const alignmentCSS = {
+          start: `left: anchor(${anchorName} left);`,
+          end: `right: anchor(${anchorName} right);`,
+          top: `top: anchor(${anchorName} top);`,
+          bottom: `bottom: anchor(${anchorName} bottom);`,
+          centerX: `left: anchor(${anchorName} center); translate: -50% 0;`,
+          centerY: `top: anchor(${anchorName} center); translate: 0 -50%;`,
+        }[alignment ?? centerAlignment];
+
+        this.anchorPositioningStyleElement.textContent = `
+          #${this.anchor} {
+            anchor-name: ${anchorName};
+          }
+          #${this.id} {
+            inset: unset;
+            ${directionCSS}
+            ${alignmentCSS}
+            position: absolute;
+          }
+        `;
```

Comment:
Would it make sense to apply just the  for adding the property to the anchor element, and then adding styles to the element with `this.$fastController.addStyles()`? Something like this:
```ts
this.anchorElement.style.setProperty('anchor-name', anchorName);

if (this.tooltipStyles) {
  this.$fastController.removeStyles(this.tooltipStyles);
}

this.tooltipStyles = css`
  #${this.id} {
    inset: unset;
    ${directionCSS ?? ''}
    ${alignmentCSS ?? ''}
    position: absolute;
  }
`;

this.$fastController.addStyles(this.tooltipStyles);
```

Reply:
This won't work with the polyfill, because `this.$fastController.addStyles()` adds styles to the constructed style sheet, so 1) the polyfill may not be able to find the style sheet to poly fill and 2) the anchor positioning related CSS declarations won't be added in unsupported browsers because `CSSStyleSheet#replaceSync()` removes unsupported declarations. See: https://github.com/oddbird/css-anchor-positioning/issues/228

Reply:
Per @marchbox's comment, we're injecting a style tag per the limitation with the polyfill.